whole-repo-tests/test_pyright.py

```diff
@@ -152,6 +153,83 @@ def test_pyright_issue_3348(tmp_path: Path):
     assert _get_pyright_errors(file) == []
 
 
+@pytest.mark.xfail(reason="Pyright doesn't resolve the numpy import")
+def test_numpy_arrays_strategy(tmp_path: Path):
+    file = tmp_path / "test.py"
+    file.write_text(
+        textwrap.dedent(
+            """
+            import numpy as np
+            from hypothesis.extra.numpy import arrays
+
+            x = arrays(dtype=np.dtype("int32"), shape=1)
+            """
+        ),
+        encoding="utf-8",
+    )
+    _write_config(tmp_path, {"typeCheckingMode": "strict"})
+    errors = _get_pyright_errors(file)
+    print(errors)
+    assert errors == []
+
+
+@pytest.mark.xfail(reason="Pyright doesn't resolve the numpy import")
```

Comment:
I mark the pyright test as xfail, because Pyright does not resolve the numpy import.
@Zac-HD, do you see how to fix it?

Reply:
Hmm, it's working locally for me, so I'm going to check how CI fails 🤔